### PR TITLE
improve CCM  API with more convience methods and a CCM::Options class

### PR DESCRIPTION
**Pull request by stdweird**
Improve CCM API by clarifying the Configuration initialisation, introduce a smarter `CacheManager->getConfiguration()` or obtain a `Configuration` instance and a `CCM::Options` class derived from `CAF::Options` to create command applications.

### CODE REVIEW

**Comment by stdweird** [comment]
This PR replace #58

File: src/main/perl/CacheManager.pm [hunk 1]
```diff
@@ -137,6 +145,167 @@ sub getCachePath
...
+
+sub getCids
+{
+    my ($self, $cid) = @_;
+
+    my $cache_path = $self->getCachePath();
+
+    my $cid_pattern = '(?:^|/)' . $PROFILE_DIR_N . '(\d+)$';
+
+    return if( ! opendir(DIR , $cache_path));
```
Comment (ned21) [inline src/main/perl/CacheManager.pm]: PBP advises against using bare file handles and recommends using indirect file handles instead.

File: src/main/perl/CacheManager.pm [hunk 2]
```diff
@@ -137,6 +145,167 @@ sub getCachePath
...
+
+    my @cids = sort map { m/$cid_pattern/; $1 } grep { m/$cid_pattern/ && -d "$cache_path/$_" } readdir(DIR);
```
Comment (ned21) [inline src/main/perl/CacheManager.pm]: This line is too long to digest. How about splitting it like this?
```
my @cids = sort map { m/$cid_pattern/; $1 } (
                               grep {m/$cid_pattern/ && -d "$cache_path/$_" } readdir(DIR));
```
or similar - using ( for continuation is python-esque, not sure if it's acceptable perl!

File: src/main/perl/CacheManager.pm [hunk 3]
```diff
@@ -137,6 +145,167 @@ sub getCachePath
...
+is held in the "latest.cid" file).
+
+=back
+
+=cut
+
+sub getCid
+{
+    my ($self, $cid) = @_;
+
```
Comment (ned21) [inline src/main/perl/CacheManager.pm]: $cid here is too close to $cids, the scalar items in @cids. How about $wantcid or $request instead?

Comment (stdweird) [inline src/main/perl/CacheManager.pm]: `$cids` is an array reference, making the scalar items `scalar @$cids`; but i renamed `$cids` to `$valid_cids` to make it more clear.

**Comment by stdweird** [comment]
@ned21 i can squash the last commits if you are ok with the changes

**Comment by ned21** [comment]
LG - thanks for addressing. I agree that squashing would be good, and then it's ready to merge.

**Comment by stdweird** [comment]
@ned21 commits squashed
@jrha i'm ok with merging this in because it's getting a bit too big, but it  might have to modify some of the public methods in `CCM::Options` (but i'll make sure it's done before 15.6 is out).

**Pull request merged**